lib/rucio/core/rse_selector.py

```diff
@@ -245,12 +255,27 @@ def __choose_rse(self, rses):
 
 
 @read_session
-def resolve_rse_expression(rse_expression, account, weight=None, copies=1, ignore_account_limit=False, size=0, preferred_rses=[], blocklist=[], prioritize_order_over_weight=False, existing_rse_size=None, *, session: "Session"):
+def resolve_rse_expression(
+    rse_expression: str,
+    account: InternalAccount,
+    weight: Optional[int] = None,
+    copies: int = 1,
+    ignore_account_limit: bool = False,
+    size: int = 0,
+    preferred_rses: Optional[list[str]] = None,
+    blocklist: Optional[list[str]] = None,
```

Comment:
`Sequence` instead of `str` (technically I think `Collection` would be better here, but we're not using the `Collection` type hint anywhere in the code at the moment, so I think it might be best to avoid it for now)

Reply:
Yeah, sorry, I meant `Sequence[str]` instead of `list[str]`

Reply:
Changed to:

```
blocklist: Optional[Sequence[str]] = None,
```